func.go

```diff
@@ -10,11 +10,16 @@ import (
 	"math"
 	"reflect"
 	"runtime"
+	"sync"
 	"unsafe"
 
 	"github.com/ebitengine/purego/internal/strings"
 )
 
+var pool = sync.Pool{New: func() interface{} {
```

Comment:
We can use `any` instead of `interface{}` since Go 1.18.

Reply:
On secont thought, we can use `interface{}` for this PR for consistency, and we can replace all `interface{}` with `any` in a later PR. So, using `any` here is an optional.